bridges/snowbridge/pallets/gas-price/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+#![cfg_attr(not(feature = "std"), no_std)]
+pub mod impls;
+pub mod types;
+
+use crate::{impls::GasPriceProvider, types::BaseFeePerGas};
+use frame_system::WeightInfo;
+pub use pallet::*;
+use sp_core::U256;
+pub const LOG_TARGET: &str = "gas-price";
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	use frame_support::pallet_prelude::*;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		GasPriceUpdate { value: U256, slot: u64 },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {}
+
+	/// Gas price
+	#[pallet::storage]
+	#[pallet::getter(fn gas_price)]
+	pub(super) type GasPrice<T: Config> = StorageValue<_, BaseFeePerGas, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {}
+}
+
+impl<T: Config> GasPriceProvider for Pallet<T> {
+	fn store(value: U256, slot: u64) {
+		<GasPrice<T>>::set(BaseFeePerGas { value, slot });
```

Comment:
I'm wondering if this pallet should provide a moving average of the gas price rather than a spot price

Reasoning:

Given that we know that the latest updated value:
1. Will only be updated every X ethereum blocks (upon receiving each finalized beacon header)
2. Will be outdated by the time an outbound message is actually delivered to Ethereum, and this is at least every 25 minutes, but will usually be 4 hours (due to BEEFY consensus delay)

So a moving average will hopefully add some buffer to account for this delay

Looking at https://stackoverflow.com/a/10157658, you could implement a moving average by blending in the latest value into an accumulator stored in the pallet.